File: Binary Search/codeforces_1929C.cpp

// Problem link ===>> https://codeforces.com/problemset/problem/1929/C
// submission link ===>> https://codeforces.com/contest/1929/submission/248492318

#include<bits/stdc++.h>
using namespace std;

int main(){
    int t;
    cin >> t;
    while(t--){
        long long k, x, a;
        cin >> k >> x >> a;

        long long aa = a;
        long long curr = 1;

        for (long long i = 1; i < x; i++)
        {
            long long next = (curr) / (k - 1);
            next++;
            curr += next;
            if (curr > a)
            {
                cout<<"NO"<<endl;
                goto next_testcase;
            }
        }

        a -= curr;
        if (a < 0)
        {
            cout << "NO" << endl;
            goto next_testcase;
        }

        a *= k;
        if (a > aa)
        {
            cout << "YES" << endl;
        }else {
            cout << "NO" << endl;
        }

    next_testcase:
        continue;
    }

    return 0;
}

File: Binary Search/codeforces_1929C.py

# Problem link ===>> https://codeforces.com/problemset/problem/1929/C

def main():
    t = int(input())

    for _ in range(t):
        k, x, a = map(int, input().split())

        aa = a
        curr = 1

        for _ in range(1, x):
            next_num = (curr // (k - 1)) + 1
            curr += next_num

            if curr > a:
                print("NO")
                break

        a -= curr
        if a < 0:
            print("NO")
        else:
            a *= k
            if a > aa:
                print("YES")
            else:
                print("NO")
    

if __name__ == "__main__":
    main()
